sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/models/traits/ProxyTrait.java

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.generic.core.models.traits;
+
+import com.generic.core.http.client.HttpClient;
+import com.generic.core.http.models.ProxyOptions;
+
+/**
+ * An {@link com.generic.core.models.traits SDK for Java trait} providing a consistent interface for configuration
+ * of proxy-specific settings. Refer to the SDK for Java
+ * <a href="https://aka.ms/azsdk/java/docs/http-client-pipeline">HTTP clients and pipelines</a> documentation for more
+ * details on proper usage and configuration of the SDK for Javan HTTP clients.
+ *
+ * @param <T> The concrete type that implements the trait. This is required so that fluent operations can continue to
+ * return the concrete type, rather than the trait type.
+ *
+ * @see com.generic.core.models.traits
+ * @see ProxyOptions
+ */
+public interface ProxyTrait <T extends HttpTrait<T>> {
```

Comment:
Is this a typo?

```suggestion
public interface ProxyTrait <T extends ProxyTrait<T>> {
```

Or are we tying `ProxyOptions` to HTTP only?

Reply:
This should be `ProxyTrait` and not `HttpTrait`.

Reply:
It's a typo, thanks for pointing it out :)